fix lost environment and long startup

mhus opened the pull request:
* Environment are lost if os.environ.clear is called for init scripts. Therefore start environment is stored.
* chown in init-home will take a long time for every start if shared home folder have lot of entries, therefore a env parameter UPDATE_HOME_WITH_UID will trigger the behaviour

xmjiao commented:
Mike, Thanks for the pull requests. I will merge some of my local changes and set up GitHub Actions for the auto testing before looking into your pull request. - Jim

xmjiao commented:
Mike, thanks for your contribution! 

I manually merged some of your fixes of the lost environment variables.

For the long startup, I changed to skip the `shared` and `project` directories when changing the owners, and also change the owners if the UID and GID on the host are different from those in Docker. Hopefully. the issue is resolved.

I tested Ubuntu 22.04 but x11vnc-0.9.14-dev failed to compile. Will need to test it again when Ubuntu 22.04 is officially released. 

Thanks again!

mhus commented:
Hi,

last days I already updated to ubuntu 22.04. I use the novnc package instead of 0.9.14 - it's broken. The newer version is currently not working in chrome browser.

I updated x11vnc-desktop, docker-desktop and vscode-desktop. I plan to use it as development environment or base for tutorials but it's not final yet. This could help you. 

Thanks for the work. Regards, Mike
